zigen-project/zen-remote-display-system
# find peer using udp broadcast

## Context

ZEN (server) has to find client (oculus) in it's LAN.

## Summary

- [x] Send UDP broadcast from client to entire LAN
- [x] Detect broadcast from client in server.
- [x] Send tcp packet from server to client to notify server presence. 

## How to check behavior

1. Build and run: https://github.com/zigen-project/zen-oculus-display-system/pull/14
Do `./gradlew clean` before build to update remote_display_system.

2. Build and run: https://github.com/zigen-project/zen/pull/173
Do `rm -rf subprojects/remote-display-system` before build to update remote_display_system.

3. You will see following log.

In Oculus Logcat,
```log
Server Found: <IP Address of the host of zen>
```

In zen's log
```log
Client Found: <IP Address of the Oculus Quest>
```
